cpp/src/arrow/adapters/orc/adapter_util.cc

```diff
@@ -15,38 +15,46 @@
 // specific language governing permissions and limitations
 // under the License.
 
+#include "arrow/adapters/orc/adapter_util.h"
+
+#include <cmath>
 #include <string>
 #include <vector>
 
-#include "arrow/adapters/orc/adapter_util.h"
 #include "arrow/array/builder_base.h"
 #include "arrow/builder.h"
+#include "arrow/chunked_array.h"
+#include "arrow/scalar.h"
 #include "arrow/status.h"
+#include "arrow/util/bitmap_ops.h"
 #include "arrow/util/checked_cast.h"
 #include "arrow/util/decimal.h"
 #include "arrow/util/range.h"
-
+#include "arrow/util/string_view.h"
+#include "arrow/visitor_inline.h"
 #include "orc/Exceptions.hh"
+#include "orc/MemoryPool.hh"
 #include "orc/OrcFile.hh"
 
+// The number of milliseconds, microseconds and nanoseconds in a second
+constexpr int64_t kOneSecondMillis = 1000LL;
+constexpr int64_t kOneMicroNanos = 1000LL;
+constexpr int64_t kOneSecondMicros = 1000000LL;
+constexpr int64_t kOneMilliNanos = 1000000LL;
+constexpr int64_t kOneSecondNanos = 1000000000LL;
 // alias to not interfere with nested orc namespace
 namespace liborc = orc;
 
-namespace arrow {
-
-namespace adapters {
-
-namespace orc {
```

Comment:
Sure.

Reply:
Hmm what exactly is the issue with ArrowOutputStream?

Reply:
On `WriteBatch`: Doc added and parameters rotated so that they make more sense.

Reply:
On the ORC error issue..let me repeat it here:

@emkornfield @pitrou It does actually look like IOError in this case. In the ORC reader liborc::ParseError is converted into IOError for IO errors and Invalid for other errors. Since this PR does not include fixing all issues in the ORC reader (which I would like to do in a latter PR after my current issues have been addressed) and that in the ORC writer there is only one instance of liborc::ParseError my current fix is adding the other two catches. However if you guys insist I can revamp error processing in the ORC reader tomorrow.